### PR TITLE
nixos/{gnome,pantheon,cinnamon,budgie}: Standardize notExcluded

### PR DESCRIPTION
**Pull request by jtojnar**
Follow up to https://github.com/NixOS/nixpkgs/pull/247036

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by bobby285271** [comment]
The `nixosTests.cinnamon` is a good way to test regressions I think as we have `programs.gnome-terminal.enable = mkDefault (notExcluded pkgs.gnome-terminal);` in the module and we have the `Open GNOME Terminal` subtest. But the test does not eval for me yet :-)

@ofborg test cinnamon

**Comment by bobby285271** [comment]
> and looks like it is no longer "notExcluded"

Ugh, to be clear, `nixosTests.cinnamon` now fails with `machine # -bash: line 1: gnome-terminal: command not found`. `notExcluded` really means "not **included** in excludePackages" :joy:

@ofborg test cinnamon

File: nixos/lib/utils.nix [hunk 1]
```diff
@@ -343,6 +343,25 @@ utils = rec {
...
+
+     Example:
+       enableByPackageName nautilus [ file-roller totem ]
+       => true
+  */
+  enableByPackageName = package: packagesToDisable:
+    let
+      namesToDisable = map getName packagesToDisable;
+    in
+      elem (getName package) namesToDisable;
```
Comment (jtojnar) [inline nixos/lib/utils.nix:363]: Ugh, should be:

```suggestion
      !elem (getName package) namesToDisable;
```

I really need find time to test it properly.

File: nixos/lib/utils.nix [hunk 2]
```diff
@@ -343,6 +343,25 @@ utils = rec {
...
+
+     Example:
+       enableByPackageName file-roller [ file-roller totem ]
+       => false
+
+     Example:
+       enableByPackageName nautilus [ file-roller totem ]
+       => true
+  */
+  enableByPackageName = package: packagesToDisable:
```
Comment (jtojnar) [inline nixos/lib/utils.nix:359]: Also the name is probably too confusing.

**Comment by jtojnar** [comment]
Fixed the negation and renamed the function to `disablePackageByName`.

Now `nix-build -A nixosTests.gnome -A nixosTests.cinnamon -A nixosTests.pantheon -A nixosTests.budgie` pass successfully.

**Review by bobby285271 — APPROVED** [review]

**Pull request merged**